### PR TITLE
feat(kyc): liveness sessions

### PR DESCRIPTION
**Pull request by danio1024**
## Summary
Added KYC Liveness Sessions.

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

File: openapi/components/schemas/KycLivenessSession.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    maxLength: 50
+    example: kyc_lvs_0YV7JMJ3DBCGRBR7K9D4HVGPP5
+  customerId:
+    $ref: ./CustomerId.yaml
+  requestId:
+    type: string
+    description: ID of the KYC request that the liveness session is associated with.
+    maxLength: 50
+    example: kyc_req_0YV7JMJ3DBCGRBR7K9D4HVGPP5
+  sessionId:
```
Comment (swilczynski) [inline openapi/components/schemas/KycLivenessSession.yaml:19]:
```suggestion
  livenessSessionId:
```

File: openapi/paths/kyc-liveness-sessions.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  x-products:
+    - Core
+  tags:
+    - KYC documents
+  summary: Create a KYC liveness session
+  operationId: PostKycLivenessSession
+  x-sdk-operation-name: create
+  description: |-
+    Creates a KYC liveness session.
+
```
Comment (lexerom) [inline openapi/paths/kyc-liveness-sessions.yaml:11]:
```suggestion
  description: Creates a KYC liveness session.
```

File: openapi/paths/storefront/kyc-liveness-sessions.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
+post:
+  x-products:
+    - Storefront
+  tags:
+    - KYC documents
+  summary: Create a KYC liveness session
+  operationId: StorefrontPostKycLivenessSession
+  x-sdk-operation-name: create
+  description: |-
+    Creates a KYC liveness session.
```
Comment (lexerom) [inline openapi/paths/storefront/kyc-liveness-sessions.yaml:10]:
```suggestion
  description: Creates a KYC liveness session.
```

File: openapi/paths/storefront/kyc-liveness-sessions.yaml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      application/json:
+        schema:
+          $ref: ../components/schemas/StorefrontKycLivenessSession.yaml
+        examples:
+          request-liveness-session:
+            summary: Basic liveness session
+            description: |-
+              This example shows how to create a liveness session with a request ID.
+            value:
+              customerId: '123_456789'
```
Comment (lexerom) [inline openapi/paths/storefront/kyc-liveness-sessions.yaml:22]:
```suggestion
              customerId: 'cus_0YV7DDSDD1C8DA64KHH2W33CPF'
```

File: openapi/paths/storefront/kyc-liveness-sessions.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    '403':
+      $ref: ../components/responses/Forbidden.yaml
+    '422':
+      $ref: ../components/responses/ValidationError.yaml
+get:
+  x-products:
+    - Storefront
+  tags:
+    - KYC documents
+  summary: Retrieve KYC liveness session
```
Comment (noon-dawg) [inline openapi/paths/storefront/kyc-liveness-sessions.yaml:53]: suggestion: I think this should be plural
```suggestion
  summary: Retrieve KYC liveness sessions
```

File: openapi/paths/storefront/kyc-liveness-sessions.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            summary: Basic liveness session
+            description: |-
+              This example shows how to create a liveness session with a request ID.
+            value:
+              customerId: 'cus_0YV7DDSDD1C8DA64KHH2W33CPF'
+              requestId: 'kyc_req_0YV7JMJ3DBCGRBR7K9D4HVGPP5'
+          no-request-liveness-session:
+            summary: Basic liveness session
+            description: |-
+              This example shows how to create a liveness session without a request ID.
```
Comment (noon-dawg) [inline openapi/paths/storefront/kyc-liveness-sessions.yaml:26]: nitpick: we should only use this format if a description has more than one sentence
```suggestion
            description: This example shows how to create a liveness session without a request ID.
```

File: openapi/paths/storefront/kyc-liveness-sessions.yaml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  requestBody:
+    content:
+      application/json:
+        schema:
+          $ref: ../components/schemas/StorefrontKycLivenessSession.yaml
+        examples:
+          request-liveness-session:
+            summary: Basic liveness session
+            description: |-
+              This example shows how to create a liveness session with a request ID.
```
Comment (noon-dawg) [inline openapi/paths/storefront/kyc-liveness-sessions.yaml:19]:
```suggestion
            description: This example shows how to create a liveness session with a request ID.
```

**Review by noon-dawg — DISMISSED** [review]
Looks good on the docs side

**Review by swilczynski — DISMISSED** [review]

**Review by noon-dawg — DISMISSED** [review]

**Review by lexerom — DISMISSED** [review]

**Review by swilczynski — DISMISSED** [review]

File: openapi/components/requestBodies/storefront/PostKycLivenessSession.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+      type: object
+      required:
+        - kycRequestId
+      properties:
+        kycRequestId:
+          type: string
+          description: ID of the KYC request that the liveness session is associated with.
+          maxLength: 50
+          example: kyc_req_0YV7JMJ3DBCGRBR7K9D4HVGPP5
+description: KYC Liveness Session.
```
Comment (noon-dawg) [inline openapi/components/requestBodies/storefront/PostKycLivenessSession.yaml:13]: nitpick: capitalization 
```suggestion
description: KYC liveness session.
```

**Review by noon-dawg — DISMISSED** [review]

File: openapi/components/schemas/StorefrontKycLivenessSession.yaml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    description: ID of the KYC liveness session.
+    maxLength: 50
+    example: kyc_liv_0YV7JMJ3DBCGRBR7K9D4HVGPP5
+  customerId:
+    $ref: ./CustomerId.yaml
+    readOnly: true
+  kycRequestId:
+    type:
+      - 'string'
+      - 'null'
```
Comment (vykhrystiuk) [inline openapi/components/schemas/StorefrontKycLivenessSession.yaml:18]:
```suggestion
    type:  'string'
```

**Review by vykhrystiuk — DISMISSED** [review]

**Review by vykhrystiuk — DISMISSED** [review]

**Review by noon-dawg — DISMISSED** [review]

File: openapi/paths/storefront/kyc-liveness-sessions@{id}.yaml [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  - $ref: ../../components/parameters/resourceId.yaml
+get:
+  x-products:
+    - Storefront
+  tags:
+    - KYC documents
+  summary: Retrieve a KYC liveness session
+  operationId: StorefrontGetKycLivenessSession
+  x-sdk-operation-name: get
+  description: Retrieves a KYC liveness session with a specified ID.
```
Comment (ashkarpetin) [inline openapi/paths/storefront/kyc-liveness-sessions@{id}.yaml:11]:
```suggestion
  description: Retrieves a KYC liveness session with a specified ID.
  security:
    - CustomerJWT: []
```

Seems custom jwt is missing here

File: openapi/paths/storefront/kyc-liveness-sessions@{id}@finish.yaml [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+          $ref: ../../components/headers/Location.yaml
+      content:
+        application/json:
+          schema:
+            $ref: ../../components/schemas/StorefrontKycLivenessSession.yaml
+    '401':
+      $ref: ../../components/responses/Unauthorized.yaml
+    '403':
+      $ref: ../../components/responses/Forbidden.yaml
+    '422':
```
Comment (ashkarpetin) [inline openapi/paths/storefront/kyc-liveness-sessions@{id}@finish.yaml:28]: I wonder in which case 422 will be thrown? What will be thrown when attempting to finish already a finished sessions?

Comment (danio1024) [reply openapi/paths/storefront/kyc-liveness-sessions@{id}@finish.yaml]: I think now it is unused, this was previously the idea, but then it was scrapped and haven't removed it.

**Review by ashkarpetin — APPROVED** [review]

**Review by swilczynski — APPROVED** [review]

**Pull request merged**